CODINGSTYLE.md

```diff
@@ -0,0 +1,42 @@
+# prplMesh coding style
+
+This document describes the intended coding style.
+Note that existing code does not always comply to it.
+
+Line length is 100 characters.
+
+## C++ code
+
+Code formatting is done with clang-format.
+See the [.clang-format](.clang-format) file for the specification.
+
+### Naming convention
+
+* Almost everything uses `lower_snake_case`.
+* `#define` uses `UPPER_SNAKE_CASE`.
+* Everything is in a namespace.
+* Therefore, there are no prefixes.
+* Private data members have an "m" prefix: `m_data`.
+* Getter functions use the name without "m": `data()`.
+* Setter functions use the "set" prefix: `set_data(value)`.
```

Comment:
what about classes and structs? They are using CamelNotation, we should add this also

Reply:
If you exclude the generated code, most classes use lower snake case.